### ISSUE TITLE
verdi status ; rabbitmq error loop?

### ISSUE THREAD

**Issue by rkovacik** [aiidateam/aiida-core#2914]
```
kovacik@iff1610:~/CODES/aiida_core 0> verdi status 
 ✓ profile:     On profile local
 ✓ repository:  /Users/kovacik/.aiida/repository-local
 ✗ postgres:    Unable to connect to iff504:5432
05/23/2019 06:17:14 PM <6424> kiwipy.rmq.communicator: [ERROR] Error starting the communicator:
Traceback (most recent call last):
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/kiwipy/rmq/communicator.py", line 535, in do_connect
    yield self._communicator.connect()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/kiwipy/rmq/communicator.py", line 363, in connect
    yield self._message_subscriber.connect()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/kiwipy/rmq/communicator.py", line 179, in connect
    self._channel = yield self._connection.channel()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/topika/connection.py", line 240, in channel
    yield channel.initialize()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/topika/robust_channel.py", line 79, in initialize
    result = yield super(RobustChannel, self).initialize()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/topika/channel.py", line 182, in initialize
    self._channel = yield self._create_channel(timeout)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/tornado/gen.py", line 307, in wrapper
    yielded = next(result)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/topika/channel.py", line 162, in _create_channel
    self._channel_maker(channel_number=self._channel_number, on_open_callback=future.set_result)
  File "/Users/kovacik/CODES/venv_aiida1.0/lib/python2.7/site-packages/pika/connection.py", line 1252, in channel
    'Channel allocation requires an open connection: %s' % self)
ConnectionWrongStateError: Channel allocation requires an open connection: <TornadoConnection CLOSED transport=None params=<ConnectionParameters host=127.0.0.1 port=5672 virtual_host=/ ssl=False>>

 ✗ rabbitmq:    Unable to connect to rabbitmq
Channel allocation requires an open connection: <TornadoConnection CLOSED transport=None params=<ConnectionParameters host=127.0.0.1 port=5672 virtual_host=/ ssl=False>>
ERROR:pika.adapters.utils.io_services_utils:Socket failed to connect: <socket._socketobject object at 0x2b90ee402ad0>; error=111 (Connection refused)
ERROR:pika.adapters.utils.connection_workflow:TCP Connection attempt failed: error(111, 'Connection refused'); dest=(2, 1, 6, '', ('127.0.0.1', 5672))
ERROR:pika.adapters.utils.connection_workflow:AMQPConnector - reporting failure: AMQPConnectorSocketConnectError: error(111, 'Connection refused')
ERROR:pika.adapters.utils.connection_workflow:AMQP connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None.
ERROR:pika.adapters.utils.connection_workflow:AMQPConnectionWorkflow - reporting failure: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None
ERROR:pika.adapters.base_connection:Full-stack connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None
ERROR:pika.adapters.base_connection:Self-initiated stack bring-up failed: AMQPConnectionError: (AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None,)
ERROR:pika.adapters.utils.io_services_utils:Socket failed to connect: <socket._socketobject object at 0x2b90ee402ad0>; error=111 (Connection refused)
ERROR:pika.adapters.utils.connection_workflow:TCP Connection attempt failed: error(111, 'Connection refused'); dest=(2, 1, 6, '', ('127.0.0.1', 5672))
ERROR:pika.adapters.utils.connection_workflow:AMQPConnector - reporting failure: AMQPConnectorSocketConnectError: error(111, 'Connection refused')
ERROR:pika.adapters.utils.connection_workflow:AMQP connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None.
ERROR:pika.adapters.utils.connection_workflow:AMQPConnectionWorkflow - reporting failure: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None
ERROR:pika.adapters.base_connection:Full-stack connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None
ERROR:pika.adapters.base_connection:Self-initiated stack bring-up failed: AMQPConnectionError: (AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - AMQPConnectorSocketConnectError: error(111, 'Connection refused'); first exception - None,)
^C
Aborted!
```
mac os x 10.9.5 `aiida_core develop branch` python2.7

**Comment by nathanielng** [comment]
I believe this may be the same issue as the following: **Unable to start job due to ConnectionWrongStateError** #2909

**Comment by sphuber** [comment]
Exactly, @rkovacik did you install RabbitMQ? Please find your [operating system instruction in the documentation](https://aiida-core.readthedocs.io/en/latest/install/prerequisites.html#ubuntu) to see how to install it.

**Comment by csadorf** [comment]
I just created a vanilla aiida environment following the [installation instructions](https://aiida-core.readthedocs.io/en/latest/install/installation.html) and when executing `verdi status`, I am exposed to the traceback similar to the one pasted in the description.

I do not think that exposing users to this traceback, especially since rabbitmq is not mentioned in the instructions and appears to be an optional dependency, is expected behavior and this issue should be classified as a bug.

**Comment by sphuber** [comment]
It is actually [explicitly marked as a pre-requisite](https://aiida-core.readthedocs.io/en/latest/install/prerequisites.html) in the documentation. I do agree however that it would be better to catch the exception and print a normal error message saying that we cannot connect to RabbitMQ

**Comment by csadorf** [comment]
> It is actually [explicitly marked as a pre-requisite](https://aiida-core.readthedocs.io/en/latest/install/prerequisites.html) in the documentation. 
You're right, I missed that.

> I do agree however that it would be better to catch the exception and print a normal error message saying that we cannot connect to RabbitMQ

I could try to tackle that if you like.

**Comment by sphuber** [comment]
Yeah, that'd be great! :+1:

**Comment by sphuber** [comment]
Fixed in #3409
